File: lib/export_m3u8.rb
require 'pathname'
require 'fileutils'

if not Pathname.new(ARGV[0]).exist?
  puts "m3u8 file not found!"
  puts "usage: #{__FILE__} path/to/playlist.m3u8"
  exit(1)
end

playlist_path   = ARGV[0]
playlist_name   = File.basename(playlist_path, ".*")

folder_path     = File.join(Dir.pwd, playlist_name)
FileUtils.mkdir_p(folder_path)

IO.read(playlist_path).split("\r").each_with_index do |file_path, index|
  next unless Pathname.new(file_path).exist?

  file_name = "#{index} " + File.basename(file_path)
  puts "Copying... #{file_name}"
  FileUtils.cp(file_path, File.join(folder_path, file_name))
end

puts "Finished."

File: lib/export_m3u8.py
#!/usr/bin/env python
# ./export_m3u8.py path/to/playlist.m3u8

import sys
import os
import shutil

if not os.path.exists(sys.argv[1]):
  print "m3u8 file not found!"
  print "usage: %s path/to/playlist.m3u8" % sys.argv[0]
  sys.exit(1)

playlist_path = sys.argv[1]
playlist_name = os.path.splitext(os.path.basename(playlist_path))[0]

folder_path   = os.path.join(os.getcwd(), playlist_name)
if not os.path.exists(folder_path):
  os.makedirs(folder_path)

playlist_file = open(playlist_path).read().split("\r")
for index, file_path in enumerate(playlist_file):
  if not os.path.exists(file_path):
    continue

  file_name = "%d %s " % (index, os.path.basename(file_path))
  print "Copying... %s" % file_name
  shutil.copy(file_path, os.path.join(folder_path, file_name))

print "Finished."
